### PR TITLE
Update release finalization automation

### PR DESCRIPTION
**Pull request by mokagio**
Brings the release finalization Fastlane automation in line with the other projects (other than for the incremental tags removal because I haven't spent the time to followup on the mobile

### Test
I think we can test these "live" when the next release finalization time comes. Everything is straightforward and the code is the same as what's used in other repos. _100% sure I just jinxed us in saying this_.

### Review
Only one developer required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -219,7 +219,15 @@ end
...
     end
 
-    trigger_app_store_ci_build(branch_to_build: "release/#{ios_get_app_version()}")
+    # Wrap up
+    version = ios_get_app_version
+    branch = "release/#{version}"
+    removebranchprotection(repository: GHHELPER_REPO, branch: branch)
+    setfrozentag(repository: GHHELPER_REPO, milestone: version, freeze: false)
+    create_new_milestone(repository: GHHELPER_REPO)
+    close_milestone(repository: GHHELPER_REPO, milestone: version)
```
Comment (mokagio) [inline fastlane/Fastfile:228]: See https://github.com/woocommerce/woocommerce-ios/blob/14dfd50e6e73f27a801c6b2aef59f9134f167c25/fastlane/Fastfile#L240-L245

As I mentioned in the PR description, the only thing I didn't add is the step to remove the incremental tags. I need to followup on that before making this PR ready for review.

**Comment by mokagio** [comment]
@oguzkocer I asked your review for this Draft PR just to let you know I'm onto it. No need to review it already if you'd rather wait for everything to be done.

**Comment by oguzkocer** [comment]
@mokagio I'd rather take a look at it all together. I've taken myself off the review list for now, but please re-add me when it's ready for review.

**Comment by mokagio** [comment]
This has become stale. Closing to keep things tidy.

**Pull request closed without merging**